man/build/build-manpages.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env -S python3 -I
+import argparse
+from collections import defaultdict
+import os
+import re
+import sys
+
+from jinja2 import Environment, FileSystemLoader
+
+def defines_in_header_file(path):
+    """Processes a C header file for lines beginning with #define,
+       extracting the symbol name and value for each one. Strips
+       double quotes from the value, if present.
+       Ignores all other lines.
+    """
+    with open(path, mode='r', encoding='UTF-8') as config_file:
+        for line in config_file:
+            # Find all lines that begin with '#define' (no leading whitespace)
+            # Use named groups to extract like:
+            # #define <name> <value> // OPTIONAL COMMENT
+            # whitespace-delimited name and value are kept; rest is discarded
+            defines = re.fullmatch(
+                r"""\#define\s+            # line must begin with #define
+                                           # followed by one or more whitespace
+                                           # characters; '#' in #define must be
+                                           # escaped because re.VERBOSE is set
+
+                    (?P<name>[^\s]+)\s+    # symbol name is second token
+                                           # symbol value is third token
+
+                    (?P<quot>['"]?)        # ... and may begin with an optional
+                                           #     single or double quote
+
+                    (?P<value>.*?)         # for the value, match any character
+                                           # but non-greedily until ...
+
+                    (?P=quot)\s*           # the character seen before the value,
+                                           # zero or more whitespace characters,
+                                           # or end of line
+
+                    (?://.*)?              # optionally strip //-style comment
+                                           # from the end of the line
+                 """,
+                line,
+                flags=re.VERBOSE
+            )
+            if defines is not None:
+                yield defines.group('name'), defines.group('value')
+
+
+def render_template_file(path, context):
+    """Creates a Jinja2 Environment object
+       and renders the template file at the provided path,
+       relative to the current working directory,
+       with the provided dict as context
+    """
+    env = Environment(
+        loader=FileSystemLoader(os.getcwd(), followlinks=True),
+        trim_blocks=True,
+        line_statement_prefix=r'.\" %'
```

Comment:
Why are you taking this approach, rather than using the stock {% if ... %} syntax? The Jinja blocks are removed by the rendering process anyway.

Reply:
Fair, somehow I was thinking that files wouldn't pass `mandoc -Tlint` if the `{% %}` syntax was used at the beginning of a line, but now that I test it… works just fine. I'll remove that quirk.

Reply:
I'd say we should run `mandoc` on the Jinja-processed _output_.